lib/secure_headers/configuration.rb

```diff
@@ -27,9 +27,11 @@ def default(&block)
           end
         end
 
-        new_config = new(&block).freeze
+        new_config = new(&block)
         new_config.validate_config!
-        @default_config = new_config
+        new_config.default_headers = new_config.generate_headers
+
+        @default_config = new_config.freeze
```

Comment:
This call to freeze makes it hard to assert that `default_headers` is getting called instead of `generate_headers` for the modified case. I'm not sure of the best way to test this...